File: Java/OOP/inheritance/Hero.java
package inheritance;

class Hero {

    String name;
    String role;
    String spesial;

    Hero(String name, String role, String spesial) {

        this.name = name;
        this.role = role;
        this.spesial = spesial;

    }

    void attack(String name) {

        System.out.printf("%s attacking %s\n",this.name, name);

    }

}

File: PHP/OOP/inheritance/Hero.php
<?php

class Hero{
    public function __construct(private $name, private $role){
    }
}

class Marksman extends Hero {
    public function __construct($name){
        $this->name = $name;
        $this->role = "Marksman";
    }
}

var_dump(new Marksman("Miya"));

